### DISCUSSION TITLE
Add `:local()` pseudo selector to SFC `<style scoped>`

### DISCUSSION THREAD

**Discussion by dpschen** [vuejs/rfcs#414] [General]
### What problem does this feature solve?
The `:global()` pseudo selector in SFC style scoped allows us to disable scoped styles for a selector.

Currently there is no option in `<style scoped>` to style something dependant on a selector that selects an element above the current component but still want to scope your styles.

My idea is to add a new `:local()` pseudo selector that makes this possible: an exception from the global scope.

#### Dark mode

Use a different color scheme depending on a `dark` class on the `<html>` element:

```vue
<style scoped>
:global(.dark) :local(.my-component) {
    color: #f2f2f2;
    background: black;
}
</style>
```

#### More general

Change the style of a component with local scoped style differently depending on a class in a parent:

```vue
<style scoped>
.card {
    border: 2px solid #ccc;
    border-radius: 10px;
    /* has no background-color */
}

:global(.has-background-image) :local(.card) {
    background-color: #f2f2f2;
}
</style>
```

### Complementing the `:deep()` selector

```vue
<style scoped>
:global(.selector-of-some-parent) :local(.selector-scoped-to-current-component) { /* [...] */}
</style>
```

Basically the same as this, just in the other tree direction:

```vue
<style scoped>
.selector-scoped-to-current-component :deep(.selector-of-some-child) { /* [...] */}
</style>
```

### Why CSS modules is no replacement

CSS modules are not compatible with SASS.

### What does the proposed API look like?
The syntax `:local()` is the same as the the selector with the same name in [CSS Modules](https://github.com/css-modules/css-modules#exceptions).

Changes are probabily mostly required inside `compiler-sfc/src/stylePluginScoped.ts`.

### Note
I created [an issue in vuejs/vue-next](https://github.com/vuejs/vue-next/issues/4948) before

**Comment by martinkruusement** [comment]
`:local()` in this case wouldn't be even necessary if simply this worked as you'd expect:

```vue
<style scoped>
:global(.theme-dark .light-surface) .title {
  color: red;
}
```

I would guess this would generate:
```css
.theme-dark .light-surface .title[data-v-00000000] {
  color: red;
}
```

The first target after the global function tagged with data-v-id.

Instead we get this:
```css
.theme-dark .light-surface {
    color: red;
}
```

Anything outside of the :global() function gets discarded.

There is a [workaround involving adding a postCSS](https://gist.github.com/tommie/c0eefee636033ff48cda50f89e3ccefa) plugin but this really should work out of the box.
